src/gui/gui-project.c

```diff
@@ -174,7 +174,7 @@ void on_projfile_add_clicked (GtkWidget* widget, void* user) {
             int amount = projectgui_list_projfiles (gummi->project->projfile);
             gtk_label_set_text (gui->projectgui->proj_nroffiles,
                                 g_strdup_printf("%d", amount));
-            gui_open_file (selected);
+            gui_open_file (selected, NULL, NULL);
```

Comment:
Root/created editor is not set -- when a new file is added.

Reply:
This would cause the preview to use the old behavior of attemping to compile the current editor/document.

Reply:
The following works when the project is newly created.

1. get all editors (and find the one that is the root, generally the first one)
2. open the file, and set the created editor's project file

```
void on_projfile_add_clicked (GtkWidget* widget, void* user) {
	gboolean status = FALSE;
	gchar* selected = NULL;
	gint rootpos;
	GuEditor* rootEditor = NULL;
	selected = get_open_filename (TYPE_LATEX);
	
	if (selected) {
		if (project_add_document (gummi->project->projfile, selected)) {
			int amount = projectgui_list_projfiles (gummi->project->projfile);
			gtk_label_set_text (gui->projectgui->proj_nroffiles,
								g_strdup_printf("%d", amount));			
			
			GList* editor_list = gummi_get_all_editors();
			gint length = g_list_length (editor_list);
			
			for (int i=0; i<length;i++) {
				GuEditor* editor = g_list_nth_data (editor_list, i);
				if(strcmp(editor->filename, gummi->project->rootfile) == 0) {
					rootEditor = editor;
					fprintf(stderr,"Found rootEditor at pos:%d\n", i);
					GuEditor* createdEditor = NULL;
					gui_open_file (selected, rootEditor, &createdEditor);
					createdEditor->projfile = gummi->project->projfile;
					rootpos = i;
					status = TRUE;
					break;
				}
			}
		}
		else {
			statusbar_set_message ("Error adding document to the project..");
		}
	}
	if (status == TRUE) projectgui_set_rootfile (rootpos);
	g_free (selected);
}
```